Submodules

rwngwn opened the pull request:
reopening #31 against master

Commit:
implementig submodule commands

Commit:
added submodule list command

Commit:
fixing unused variables errors

Commit:
do not checkout common steps and features by default - its not needed when containers are not tested

Commit:
implemented remove command for submodules

vrutkovs commented:
+1 on the code, tests need some adjustment though - we should add existing common-steps/common-steps before we do `update`

Commit:
changed update command to update or remotes
had to remove sanity check(probably is better to move it to some subcommand), because there should not been some features/steps defined in time it is run

rwngwn commented:
@vrutkovs fixed tests and added common steps back.

Commit:
updating docs

vrutkovs commented on ctf_cli/application.py:
```diff
@@ -100,22 +101,13 @@ def init(self):
             logger.info("Directory tests/steps/common_steps already exists")
         else:
             logger.info("Adding tests/steps/common_steps as a submodule")
             check_call('git submodule add https://github.com/Containers-Testing-Framework/common-steps.git tests/steps/common_steps', shell=True)
```
We shouldn't forcibly add `common-steps` here

vrutkovs commented on circle.yml:
```diff
@@ -15,6 +15,7 @@ test:
   override:
     - mkdir test_dir
     - cd test_dir && COVERAGE_FILE=../.coverage.init coverage run ../ctf-cli.py init
+    - cd test_dir && COVAREGE_FILE=../.coverage.remote_add coverage run ../ctf-cli.py remote add features https://github.com/Containers-Testing-Framework/common-features.git
```
We should add `common-steps` here instead

rwngwn commented:
hmm I cant do that easily because python doesn't like - character in imports it seems. so if I add common-steps this way. it dies.
@vrutkovs can you change repo name to common_steps instead of common-steps ? :)

Commit:
make submodule dir repo name and removed common steps form downloading by default

vrutkovs commented:
`_write_environment_py` from `behave.py` should be updated - e.g. convert dashes into underscores.

Alternatively we can add a submodule with a new name (stripping non-alphanumeric chars)

rwngwn commented:
@vrutkovs Ok.. will update it in _write_environment_py. you ar right its better solution

Commit:
change - to _ in repositories name for python import to work
